From a41779142fef8c1f41f6d1c25a3fcef098d13369 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Fri, 12 Oct 2018 23:09:22 +0800
Subject: [PATCH 4831/5242] MLK-19884: soc: imx: sc: main: ipc: fix onoff key
 can't wakeup kernel from 'freeze' mode

commit  b590aa8846c9b6f1925e8ef5c67fd51572779938 from
https://source.codeaurora.org/external/imx/linux-imx.git

That caused by commit d5c4a949522f6 ("MLK-19515 soc: imx: ipc: remove
unnecessary set wake for mu") which remove irq_set_irq_wakeup directly.
Actually, we can keep that in driver without set_irq_wake() interface in
GIC driver if cherry-pick commit 4110b5cbb014e ("irqchip/gic-v3: Allow
interrupt to be configured as wake-up sources") from upstreaming kernel.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/sc/main/ipc.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/imx/sc/main/ipc.c b/drivers/soc/imx/sc/main/ipc.c
index fd888a0..f795b24 100644
--- a/drivers/soc/imx/sc/main/ipc.c
+++ b/drivers/soc/imx/sc/main/ipc.c
@@ -344,6 +344,8 @@ int __init imx8_mu_init(void)
 					irq, err);
 			return err;
 		}
+
+		irq_set_irq_wake(irq, 1);
 	}
 
 	if (!scu_mu_init) {
-- 
1.7.9.5

